src/main/java/httpserver/httpresources/MethodNotAllowed.java

```diff
@@ -0,0 +1,14 @@
+package httpserver.httpresources;
+
+import httpserver.HttpResponse;
+import httpserver.Resource;
+
+public class MethodNotAllowed implements Resource {
+    @Override
+    public HttpResponse act() {
+        HttpResponse response = new HttpResponse();
+        response.responseLine += "405 Method Not Allowed";
+        response.headers.add("Allow: HEAD, OPTIONS");
+        return response;
```

Comment:
Since classes cannot have public instance variables, we will need to find a different way to put the object in the right state.

Reply:
I included setter and getter methods.  Does that sound okay?

Reply:
Sounds good to me. Do we need setters? Or can we use the constructor?

Reply:
Either could work.  The benefit of setters is we only have to enter applicable information.

Reply:
I see. One of the downsides of setters is that it is easier to put your object into an inconsistent state. There is a OO pattern that marries the two approachs (configuring via constructor vs configuring via setters) called the [Builder pattern](https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/)

```java
HttpResponse response 
  = new HttpResponse.Builder()
       .status(200)
       .addHeader(header)
       .body("Hello world")
       .build();

```

Reply:
Thank you!  Will incorporate.